canonicalwebteam/discourse/parsers/base_parser.py

```diff
@@ -286,18 +300,48 @@ def _parse_metadata(self, index_soup, section_name):
                     if value.find("a"):
                         row_dict["topic_name"] = value.find("a").text
 
-                    # Beautiful soup renders URLs as anchors
-                    # Avoid that default behaviour
-                    if value.find("a") and (
-                        value.find("a")["href"] == value.find("a").text
-                    ):
-                        value.contents[0] = value.find("a").text
+                        # Only engage pages need a link
+                        if value.findAll("a", href=True):
+                            if value.find("a")["href"] == value.find("a").text:
+                                value.contents[0] = value.find("a").text
+
+                        else:
+                            error = MissingLinkError(
+                                f"Warning: Link not found when parsing row {index} \"{row_dict['topic_name']}\"\
+                                {titles[index]}. \
+                                This Engage page has been skipped."
+                            )
+                            flask.current_app.extensions[
+                                "sentry"
+                            ].captureMessage(error)
+                            self.metadata_errors.append(error)
+                            row_dict = None
+                            break
+
+                    # Missing path will cause the engage item in index to not
+                    # link to the corresponding page
+                    # Missing type will cause resource_name to be empty in
+                    # thank-you pages
+                    # This error does not need breaking, because it does not
+                    # break the page
+                    if (
+                        (titles[index] == "path") or (titles[index] == "type")
+                    ) and ((value.text == "") or (value.text is None)):
+                        error = MissingTitleError(
+                            f"Warning: row {index} \"{row_dict['topic_name']}\"\
+                            {titles[index]} is missing. This Engage page has \
+                            been skipped."
```

Comment:
Help me understand what this error message will actually say? Is it something like:

> Warning: row 12 "OpenStack survey" path is missing. This Engage page has been skipped.

I think maybe this could be improved to something like:

> Warning: In row 12 ("OpenStack survey"), the "path" field is missing. This row has been skipped.